pyproj/_crs.pyx

```diff
@@ -117,15 +244,165 @@ cdef class AreaOfUse:
                 &area_of_use.north,
                 &area_name):
             return None
-        area_of_use.name = pystrdecode(area_name)
+        area_of_use.name = decode_or_undefined(area_name)
         return area_of_use
 
     @property
     def bounds(self):
         return self.west, self.south, self.east, self.north
 
 
-cdef class Ellipsoid:
+cdef class Base:
+    def __cinit__(self):
+        self.projobj = NULL
+        self.projctx = NULL
+        self.name = "undefined"
```

Comment:
Should we have a typed definition of those attributes? 

Like 

```
cdef class Base:
    cdef PJ *projobj
    cdef PJ_CONTEXT *projctx
...
```

(I don't know enough from cython to know what benefit this gives, as the current code is also working, but the example I give (and addition to initializing them in (c)init ) is how they show it in the docs)

Reply:
it is typed in the `_crs.pxd` file.